src/CloudEvents/V1/CloudEvent.php

```diff
@@ -145,4 +164,47 @@ public function setTime(?DateTimeInterface $time): CloudEvent
 
         return $this;
     }
+
+    /**
+     * @return array<string,bool|int|string>
+     */
+    public function getExtensions(): array
+    {
+        return $this->extensions;
+    }
+
+    /**
+     * @param bool|int|string|null $value
+     */
+    public function setExtension(string $attribute, $value): CloudEvent
+    {
+        if (\preg_match('/^[a-z0-9]+$/', $attribute) !== 1) {
+            throw new ValueError(
+                \sprintf('%s(): Argument #1 ($attribute) must match the regex [a-z0-9]+, %s given', __METHOD__, $attribute)
```

Comment:
NB The spec says MUST match `[a-z0-9]+` and SHOULD be no longer than 20 characters.